tests/migration_test.go

```diff
@@ -4194,25 +4141,97 @@ func temporaryTLSConfig() *tls.Config {
 	}
 }
 
-func disableNodeLabeller(node *k8sv1.Node, virtClient kubecli.KubevirtClient) *k8sv1.Node {
+func stopNodeLabeller(nodeName string, virtClient kubecli.KubevirtClient) *k8sv1.Node {
 	var err error
+	var node *k8sv1.Node
 
-	By(fmt.Sprintf("Patching node to %s include %s label", node.Name, v1.LabellerSkipNodeAnnotation))
+	suiteConfig, _ := GinkgoConfiguration()
+	Expect(suiteConfig.ParallelTotal).To(Equal(1), "stopping / resuming node-labeller is supported for serial tests only")
+
+	By(fmt.Sprintf("Patching node to %s include %s label", nodeName, v1.LabellerSkipNodeAnnotation))
 	key, value := v1.LabellerSkipNodeAnnotation, "true"
-	libnode.AddAnnotationToNode(node.Name, key, value)
+	libnode.AddAnnotationToNode(nodeName, key, value)
 
-	By(fmt.Sprintf("Expecting node %s to include %s label", node.Name, v1.LabellerSkipNodeAnnotation))
+	By(fmt.Sprintf("Expecting node %s to include %s label", nodeName, v1.LabellerSkipNodeAnnotation))
 	Eventually(func() bool {
-		node, err = virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+		node, err = virtClient.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		value, exists := node.Annotations[v1.LabellerSkipNodeAnnotation]
+		return exists && value == "true"
+	}, 30*time.Second, time.Second).Should(BeTrue(), fmt.Sprintf("node %s is expected to have annotation %s", nodeName, v1.LabellerSkipNodeAnnotation))
+
+	return node
+}
+
+func resumeNodeLabeller(nodeName string, virtClient kubecli.KubevirtClient) *k8sv1.Node {
```

Comment:
I am wondering why we need to wait for the handler. Couldn't we just add labels that we remove/remove that we add and so on?

Reply:
So we would ensure we have a clean testing environment.
This isn't different from what's being done with many functions in `BeforeTestCleanup()` for example.

As we have experienced before in these tests, backing up & restoring node labels manually is much more error prone. In addition, the overhead is negligible; we're talking about a matter of seconds for a very small number of tests.